insights/parsers/uname.py

```diff
@@ -210,13 +209,13 @@ def parse_content(self, content):
             raise UnameError('Empty uname line', '')
 
         data = {}
-        uname_line = content[-1]  # read the last line instead of the first
+        uname_list = [line for line in content if line.startswith("Linux")]
+        if not uname_list:
+            raise UnameError("Uname string appears invalid", '')
+        uname_line = uname_list[0]
         uname_parts = uname_line.split(' ')
         if len(uname_parts) < 3:
-            ver_rel_match = re.match("[0-9](\.[0-9]+){2}-[0-9]+", uname_parts[0])
-            if not ver_rel_match:
-                raise UnameError("Uname string appears invalid", uname_line)
-            data['kernel'] = uname_parts[0]
+            raise UnameError("Uname string appears invalid", uname_line)
```

Comment:
@wushiqinlou this code is inherited from the original `uname` utility class as is the classmethod `from_kernel` below.  It does allow the creation of uname objects from a shortened uname string.  I'm not convinced that we want to remove this working code unless it is causing problems with the parsing.